native/e2e/record.js

```diff
@@ -0,0 +1,47 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+const EventEmitter = require('events');
+const { cleanFileName } = require('./utils');
+
+const AndroidRecorder = require('./recorders/android-recorder');
+const IOSRecorder = require('./recorders/ios-recorder');
+
+const Recorder =
+  process.env.BPK_PLATFORM === 'android' ? AndroidRecorder : IOSRecorder;
+
+module.exports = storybookController => async (
+  kind,
+  story,
+  basePath,
+  testFuction,
+) => {
+  const videoName = cleanFileName(story, kind);
+  const recorder = new Recorder(`${basePath}/${videoName}`, videoName);
+  const testEmitter = new EventEmitter();
+
+  testEmitter.on('test:started', () => recorder.startRecording());
```

Comment:
Aren't event names like these usually in present tense? e.g `test:start` and `test:end`?

Reply:
there are all sort of discussion about this, the side I've picked comes from DDD, an event is a reaction to something that happened, so the callback runs when the action has already been executed (in this case the test started), we might use events in the future so it'lll be good to make a decision on this, toughts? @shaundon @tiagoengel @amolgupta @georgegillams @matthewdavidson

Reply:
With all Streams you have `on('data')` and `on('end')`